File: frontend/src/types/application.ts
import { Vacancy } from "./vacancy";
import { User } from "./auth";

export enum ApplicationStatus {
  PENDING = "pending",
  VIEWED = "viewed",
  ACCEPTED = "accepted",
  REJECTED = "rejected",
}

export interface Application {
  id: number;
  status: ApplicationStatus;
  created_at: string;
  updated_at?: string;
  cover_letter?: string;
  vacancy: Vacancy;
  applicant?: User; // Available for employer
  resume_id?: number;
  resume_title?: string;
}

File: backend/app/models/application.py
from sqlalchemy import Column, Integer, String, ForeignKey, Enum, Text, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.core.database import Base
import enum

class ApplicationStatus(str, enum.Enum):
    PENDING = "pending"
    VIEWED = "viewed"
    ACCEPTED = "accepted"
    REJECTED = "rejected"

class Application(Base):
    __tablename__ = "applications"

    id = Column(Integer, primary_key=True, index=True)
    vacancy_id = Column(Integer, ForeignKey("vacancies.id"), nullable=False)
    applicant_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    
    status = Column(Enum(ApplicationStatus), default=ApplicationStatus.PENDING, nullable=False)
    cover_letter = Column(Text, nullable=True)
    resume_id = Column(Integer, ForeignKey("resumes.id"), nullable=True)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    vacancy = relationship("Vacancy", back_populates="applications")
    applicant = relationship("User", back_populates="applications")
    resume = relationship("Resume", back_populates="applications") # Assuming Resume model exists or will exist

# Add to User model
# User.applications = relationship("Application", back_populates="applicant")
